Topic0_4/src/DBBuilder/MySQLConnectionBuilder.java

```diff
@@ -0,0 +1,29 @@
+package DBBuilder;
+
+public class MySQLConnectionBuilder implements DBConnectionBuilder {
+	
+	private DBConnection connection;
+	
+	public MySQLConnectionBuilder() {
+			connection = new DBConnection();
```

Comment:
since your object is created at first place and it's always accessible using getDBConnection method I think this is not a valid builder. A valid builder makes sure the object is built with a correct state, for example it doesn't allow you to get object without the user or password already set.